src/workspaces/features/RunPipelineDialog/RunPipelineDialog.tsx

```diff
@@ -49,31 +76,32 @@ const RunPipelineDialog = (props: RunPipelineDialogProps) => {
       clearCache();
       onClose();
     },
-    getInitialState() {
-      const version =
-        ("version" in props && props.version) ||
-        ("run" in props && props.run.version) ||
-        pipeline.currentVersion ||
-        null;
-
-      let state: any = {
-        version,
-      };
-
-      if ("run" in props && props.run) {
-        state = { ...state, ...props.run.config };
-      }
-
-      return state;
-    },
   });
   const { t } = useTranslation();
 
   useEffect(() => {
     if (!open) {
       form.resetForm();
+      setShowVersionPicker(false);
+    } else if (!("version" in props)) {
+      fetch({
+        variables: {
+          workspaceSlug: pipeline.workspace?.slug,
+          pipelineCode: pipeline.code,
+        },
+      });
     }
-  }, [open, form]);
+  }, [open, form, fetch, props, pipeline]);
+
+  useEffect(() => {
+    if ("version" in props && props.version) {
```

Comment:
You can remove the version from the props. This is not yet used anywhere and make this more complex